sanic/request.py

```diff
@@ -127,14 +108,32 @@ def __repr__(self):
             self.__class__.__name__, self.method, self.path
         )
 
-    def body_init(self):
```

Comment:
See comment on [forums](https://community.sanicframework.org/t/new-streaming-api-and-other-major-changes/518/8?u=ahopkins)

Reply:
Deprecated in 20.3 patch already merged.

This PR now raises `NotImplementedError` if a custom request class is used where any of the body functions are implemented but `receive_body` isn't. For cross-version compatibility, it is allowed to implement both APIs.